specification/agricultureplatform/AgriculturePlatform.DataPlane/models.tsp

```diff
@@ -0,0 +1,2140 @@
+   */
+  @minLength(1)
+  provider: string;
+
+  /**
+   * ItemUri.
+   */
+  @minLength(1)
+  itemUri: string;
+
+  /**
+   * Data Model for SatelliteIngestionJobRequest.
+   */
+  data: SatelliteData;
+
+  /**
+   * Collection of satellite data. Available Value: sentinel-2-l2a, sentinel-2-l1c.
+   */
+  @minLength(1)
+  providerCollectionId: string;
+
+  /**
+   * Feature Id.
+   */
+  featureId?: string;
+}
+
+/**
+ * OAuthClientCredentials for clientId clientSecret auth.
+ */
+// #suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "For backward compatibility"
+// @discriminator("kind")
+model OauthClientCredentials extends AuthCredentials {
+  /**
+   * ClientId associated with the provider.
+   */
+  @minLength(1)
+  clientId: string;
+
+  /**
+   * Properties of the key vault.
+   */
+  clientSecret: KeyVaultProperties;
+
+  /**
+   * Type of AuthCredentials.
+   */
+  kind: AuthCredentialsKind.OAuthClientCredentials;
+}
+
+/**
+ * Properties of the key vault.
+ */
+model KeyVaultProperties {
+  /**
+   * Uri of the key vault.
+   */
+  @maxLength(100)
+  @minLength(2)
+  keyVaultUri: string;
+
+  /**
+   * Name of Key Vault key.
+   */
+  @maxLength(256)
+  @minLength(2)
+  keyName: string;
+
+  /**
+   * Version of Key Vault key.
+   */
+  @maxLength(256)
+  @minLength(2)
+  keyVersion: string;
+}
+
+/**
+ * AuthCredentials abstract base class for Auth Purpose.
+ */
+@discriminator("kind")
+model AuthCredentials {
+  /**
+   * Enum for different types of AuthCredentials supported.
+   */
+  kind: AuthCredentialsKind;
+}
+
+/**
+ * Resource model for Satellite API.
+ */
+@resource("external/satellite")
+model SatelliteResource {
+  /**
+   * Dummy key for satellite resource
+   */
+  @key("key")
+  @visibility("read")
+  key: string;
+}
+/**
+ * Data Model for SatelliteIngestionJobRequest.
+ */
+model SatelliteData {
+  /**
+   * List of ImageNames.
+   */
+  imageNames?: string[];
+
+  /**
+   * List of ImageFormats. Available value: TIF.
+   */
+  imageFormats?: string[];
+
+  /**
+   * List of ImageResolutions in meters. Available values: 10, 20, 60.
+   */
+  imageResolutions?: float64[];
+}
+
+/**
+ * Satellite Fetch Response.
+ */
+model SatelliteFetchResponse {
+  /**
+   * Value.
+   */
+  value?: SatelliteFetchData[];
+}
+
+/**
+ * SatelliteFetchData.
+ */
+model SatelliteFetchData {
+  /**
+   * Start Date.
+   */
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  dateTime?: utcDateTime;
+
+  /**
+   * Provider of satellite data. Available Value: SentinelHub.
+   */
+  provider?: string;
+
+  /**
+   * Collection of satellite data. Available Value: sentinel-2-l2a, sentinel-2-l1c.
+   */
+  @minLength(1)
+  providerCollectionId: string;
+
+  /**
+   * Feature Id.
+   */
+  featureId?: string;
+
+  /**
+   * ImageFiles.
+   */
+  imageFiles?: ImageFile[];
+
+  /**
+   * ItemUri.
+   */
+  itemUri?: string;
```

Comment:
Could this be of type `url`? https://typespec.io/docs/standard-library/built-in-data-types/#url